crates/service/src/utils/dir_ops.rs

```diff
@@ -1,7 +1,25 @@
-use std::fs::File;
+use std::fs::{read_dir, File};
+use std::io;
 use std::path::Path;
 
 pub fn sync_dir(path: impl AsRef<Path>) {
     let file = File::open(path).expect("Failed to sync dir.");
     file.sync_all().expect("Failed to sync dir.");
 }
+
+pub fn dir_size(dir: &Path) -> io::Result<usize> {
+    let mut size = 0;
+    if dir.is_dir() {
+        for entry in read_dir(dir)? {
+            let entry = entry?;
+            let path = entry.path();
+            let name = path.file_name().unwrap().to_string_lossy();
+            if path.is_dir() && !name.starts_with('.') {
```

Comment:
Why `!name.starts_with('.')`?

Reply:
Ignore the dir starts with `.`.

Reply:
But isn't it not ignored in fact? It goes into `else` statement.

Reply:
Fixed.